third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -798,6 +805,27 @@ impl<'a> FunctionGenerator<'a> {
         }
     }
 
+    /// If a temp already on the abstract stack is both:
+    ///   - not a source of the current instruction
+    ///   - destination of the current instruction
+    /// then, we have a conflicting write to that temp.
+    ///
+    /// This method ensures that conflicting writes do not happen by flushing out such temps
+    /// from the abstract stack before emitting code for the current instruction.
+    fn flush_any_conflicts(
+        &mut self,
+        ctx: &BytecodeContext,
+        dests: &[TempIndex],
+        sources: &[TempIndex],
+    ) {
+        let dests = BTreeSet::from_iter(dests.iter());
+        let sources = BTreeSet::from_iter(sources.iter());
+        let conflicts = dests.difference(&sources).collect::<BTreeSet<_>>();
+        if let Some(pos) = self.stack.iter().position(|t| conflicts.contains(&t)) {
+            self.abstract_flush_stack_before(ctx, pos);
```

Comment:
I wonder why coverage warning shows 825 isn't covered?

Reply:
As I had mentioned in a previous iteration of this PR, reproducing this bug is difficult, unless we keep around other bugs. There were two other bug fixes/improvements that were merged in: variable coalescing and its interaction with certain dead stores (https://github.com/aptos-labs/aptos-core/pull/12571), improved transitive dead store elimination (https://github.com/aptos-labs/aptos-core/pull/12357). With both of these merged in, unfortunately the tests I added here do not reproduce the bug (I had tested the reproduction independently before one or more of those were in).

Based on the issue this PR addresses, I have created these two simple commits to:
1. Showcase the bug: https://github.com/aptos-labs/aptos-core/commit/a634e036dc510f97a480197ffb91cce5ad8b7497. See comment at the end on how to reproduce it.
2. The fix provided in this PR actually does fix the reproducible bug above: https://github.com/aptos-labs/aptos-core/commit/1800749c79b0858cbde1bfc064b7d0cbcc2b9453

Handcrafting the stackless bytecode is one option if we really want to keep around a test that is not fickle, but that is too cumbersome right now, and may only be relevant until we fix many other things/rewrite file format generator.

For now, not taking any further action because of reasons provided above.

Reply:
Yeah, an "asm/mvir" primitive that stitches in some specified stackless bytecode at bytecode gen time might be neat, but handling types would make it really hard to implement.